### PR TITLE
EOS-17213 : Migration to "message_bus" as default setting

### PR DESCRIPTION
**Pull request by pranavPawar07**
-----
[View rendered s3backgrounddelete/rabbitmq_setup.md](https://github.com/Seagate/cortx-s3server/blob/br/pranav/EOS-17213/s3backgrounddelete/rabbitmq_setup.md)
[View rendered s3backgrounddelete/readme.md](https://github.com/Seagate/cortx-s3server/blob/br/pranav/EOS-17213/s3backgrounddelete/readme.md)

### CODE REVIEW

File: ansible/setup_backgrounddelete_config.sh [hunk 1]
```diff
@@ -22,14 +22,8 @@ set -e
...
 #
-# As of now, s3backgrounddelete uses rabbitmq server to communicate
-# message between producer and consumer. So we need to configure Rabbitmq
-# on the system. Refer: <cortx-s3server>/s3backgrounddelete/rabbitmq_setup.md
-
-# Install rabbit-mq.
-# As of now rabbit-mq is being installed as part of motr dependency installation.
-# curl -s https://packagecloud.io/install/repositories/rabbitmq/rabbitmq-server/script.rpm.sh | sudo bash
-
+# s3backgrounddelete uses message bus to communicate
```
Comment (amitwac2608) [inline ansible/setup_backgrounddelete_config.sh:25]: cortx message bus

File: s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py [hunk 2]
```diff
@@ -244,95 +244,14 @@ def get_daemon_mode(self):
...
-            raise KeyError(
-                "Could not parse rabbitmq durable from config file " +
-                self._conf_file)
-
     def get_schedule_interval(self):
-        """Return schedule interval of rabbitmq from config file or KeyError."""
+        """Return schedule interval of object recovery scheduler from config file or KeyError."""
         try:
-          schedule_interval = self.s3confstore.get_config('rabbitmq>schedule_interval_secs')
+          schedule_interval = self.s3confstore.get_config('message_bus>scheduler_schedule_interval')
```
Comment (amitwac2608) [inline s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py:250]: move this key as constant to cortx_s3_constants file.

File: s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py [hunk 3]
```diff
@@ -18,13 +18,13 @@
 #
 
 """
-This class act as object recovery processor which consumes
-the rabbitmq message queue.
+This class acts as object recovery processor which consumes
+the message_bus message queue.
```
Comment (amitwac2608) [inline s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py:22]: cortx message_bus

File: s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py [hunk 4]
```diff
@@ -50,29 +50,16 @@ def consume(self):
...
-                    self.config.get_rabbitmq_host(),
-                    self.config.get_rabbitmq_exchange(),
-                    self.config.get_rabbitmq_queue_name(),
-                    self.config.get_rabbitmq_mode(),
-                    self.config.get_rabbitmq_durable(),
-                    self.logger)
             else:
                 self.logger.error(
-                "Invalid argument specified in messaging_platform use message_bus or rabbit_mq")
+                "Invalid argument specified in messaging_platform use 'message_bus'")
```
Comment (amitwac2608) [inline s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py:62]: log the value as well, and do not say to use 'message_bus'. Tomorrow we may use something else... just log the value and say that this value is not supported.

File: s3backgrounddelete/s3backgrounddelete/config/s3_background_delete_config.yaml.sample [hunk 5]
```diff
@@ -37,6 +37,7 @@ message_bus:
    consumer_sleep: 5
    purge_sleep: 60
    admin_id: "admin_s3_background_delete"
+   scheduler_schedule_interval: 1200                        # Schedule Interval is time period at which object recovery scheduler will be executed (in seconds). Increasing this value leads to reduced memory consumption.
```
Comment (nileshgovande) [inline s3backgrounddelete/s3backgrounddelete/config/s3_background_delete_config.yaml.sample:40]: It should be in cortx_s3:

File: ansible/setup_backgrounddelete_config.sh (hunk 1 above)
Comment (pranavPawar07) [reply ansible/setup_backgrounddelete_config.sh]: done

File: s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py (hunk 3 above)
Comment (pranavPawar07) [reply s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py]: done

File: s3backgrounddelete/s3backgrounddelete/config/s3_background_delete_config.yaml.sample (hunk 5 above)
Comment (pranavPawar07) [reply s3backgrounddelete/s3backgrounddelete/config/s3_background_delete_config.yaml.sample]: done

File: s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py (hunk 4 above)
Comment (pranavPawar07) [reply s3backgrounddelete/s3backgrounddelete/object_recovery_processor.py]: done.

**Comment by pranavPawar07** [comment]
S3 premerge job : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/603/

**Review by nileshgovande — APPROVED** [review]

**Review by vimalraghwani20 — APPROVED** [review]

**Pull request merged**